diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 4300c27b..43cea5d9 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -63,11 +63,8 @@ else()
     set(runtime_target ${CMAKE_SYSTEM_NAME}/$<CONFIG>)
 endif()
 
-install(TARGETS realm-wrappers
-    RUNTIME DESTINATION ${runtime_target}
-    LIBRARY DESTINATION ${runtime_target}
-    FRAMEWORK DESTINATION ${runtime_target}
-)
+install(TARGETS realm-wrappers)
+
 if(MSVC)
     install(FILES $<TARGET_PDB_FILE:realm-wrappers> DESTINATION ${runtime_target} OPTIONAL)
 endif()
\ No newline at end of file
