diff --git a/CMakeLists.txt b/CMakeLists.txt
index c649b40f1..76f644c6b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -222,7 +222,6 @@ elseif(CMAKE_SYSTEM_NAME MATCHES "Linux|Android")
 endif()
 
 if(REALM_NEEDS_OPENSSL OR REALM_FORCE_OPENSSL)
-    set(OPENSSL_USE_STATIC_LIBS ON)
     if(VCPKG_TOOLCHAIN)
         # If we're building with vcpkg, prefer to find OpenSSL there first
         find_package(OpenSSL)
