diff --git a/setup.py b/setup.py
index 1c093d5..255caba 100644
--- a/setup.py
+++ b/setup.py
@@ -16,7 +16,7 @@ with open(path.join(here, 'steam/__init__.py'), encoding='utf-8') as f:
 
 install_requires = [
     'six>=1.10',
-    'pycryptodomex>=3.7.0',
+    'pycryptodome>=3.7.0',
     'requests>=2.9.1',
     'vdf>=3.3',
     'cachetools>=3.0.0',
diff --git a/steam/core/crypto.py b/steam/core/crypto.py
index 6fb2867..b55c597 100644
--- a/steam/core/crypto.py
+++ b/steam/core/crypto.py
@@ -6,10 +6,10 @@ from os import urandom as random_bytes
 from struct import pack
 from base64 import b64decode
 
-from Cryptodome.Hash import SHA1, HMAC
-from Cryptodome.PublicKey.RSA import import_key as rsa_import_key, construct as rsa_construct
-from Cryptodome.Cipher import PKCS1_OAEP, PKCS1_v1_5
-from Cryptodome.Cipher import AES as AES
+from Crypto.Hash import SHA1, HMAC
+from Crypto.PublicKey.RSA import import_key as rsa_import_key, construct as rsa_construct
+from Crypto.Cipher import PKCS1_OAEP, PKCS1_v1_5
+from Crypto.Cipher import AES as AES
 
 
 class UniverseKey(object):
