From dc176aee43de74de73cfa9f9dbd1b467ff7ad57b Mon Sep 17 00:00:00 2001
From: Tilmann Meyer <me@atiltedtree.dev>
Date: Sun, 13 Feb 2022 12:36:20 +0100
Subject: [PATCH 1/4] meson: add chromaprint option

---
 meson.build       | 2 +-
 meson_options.txt | 6 ++++++
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index a58af0956..8400c057e 100644
--- a/meson.build
+++ b/meson.build
@@ -1070,7 +1070,7 @@ foreach func : fftw3_funcs
 endforeach
 
 chromaprint_dep = dependency (
-  'libchromaprint', required: false,
+  'libchromaprint', required: get_option ('chromaprint'),
   static: all_static)
 if (chromaprint_dep.found ())
   cdata.set('HAVE_CHROMAPRINT', 1)
diff --git a/meson_options.txt b/meson_options.txt
index c6f4d8502..c36cdd759 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -87,6 +87,12 @@ option (
   value: 'auto',
   description: 'Build with LSP DSP support for optimized DSP')
 
+option (
+  'chromaprint',
+  type: 'feature',
+  value: 'auto',
+  description: 'Build with chromaprint support')
+
 option (
   'static_lsp_dsp',
   type: 'boolean',
-- 
2.35.1

