From 4d8d12970c79e3da7c6113e9e1ad68d6291206cc Mon Sep 17 00:00:00 2001
From: Tilmann Meyer <me@atiltedtree.dev>
Date: Sun, 13 Feb 2022 12:37:07 +0100
Subject: [PATCH 3/4] backtrace: only use execinfo on glibc

---
 src/utils/backtrace.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/utils/backtrace.c b/src/utils/backtrace.c
index 05dea8830..478f722c6 100644
--- a/src/utils/backtrace.c
+++ b/src/utils/backtrace.c
@@ -44,7 +44,7 @@
 #  include <windows.h>
 
 #  include <dbghelp.h>
-#else
+#elif __GLIBC__
 #  include <execinfo.h>
 #endif
 
@@ -435,7 +435,7 @@ read_traditional_bt:
     }
   free (symbol);
   free (line);
-#else
+#elif __GLIBC__
   void *  array[max_lines];
   char ** strings;
 
-- 
2.35.1

