From fb6322542e7d2d0361e3242c06eff0a5dfe11b2b Mon Sep 17 00:00:00 2001
From: Tilmann Meyer <me@atiltedtree.dev>
Date: Sun, 13 Feb 2022 12:37:07 +0100
Subject: [PATCH 3/4] backtrace: remove execinfo, make it musl compatible

---
 src/utils/backtrace.c | 26 --------------------------
 1 file changed, 26 deletions(-)

diff --git a/src/utils/backtrace.c b/src/utils/backtrace.c
index c36aefae0..c8b1f2f2b 100644
--- a/src/utils/backtrace.c
+++ b/src/utils/backtrace.c
@@ -43,8 +43,6 @@
 #ifdef _WOE32
 #include <windows.h>
 #include <dbghelp.h>
-#else
-#include <execinfo.h>
 #endif
 
 #include "utils/backtrace.h"
@@ -446,30 +444,6 @@ read_traditional_bt:
     }
   free (symbol);
   free (line);
-#else
-  void * array[max_lines];
-  char ** strings;
-
-  /* get void*'s for all entries on the stack */
-  int size = backtrace (array, max_lines);
-
-  /* print out all the frames to stderr */
-  strings = backtrace_symbols (array, size);
-  for (int i = 0; i < size - 2; i++)
-    {
-      /* if addr2line failed, print what we can */
-      if (!with_lines ||
-          addr2line (
-            exe_path, array[i],
-            current_line,
-            size - 2 - i - 1) != 0)
-        {
-          sprintf (
-            current_line, "%s\n", strings[i]);
-        }
-      strcat (message, current_line);
-    }
-  free (strings);
 #endif
 
   return g_strdup (message);
-- 
2.35.1

