diff --git a/meson.build b/meson.build
index 407fd1ddea..eaf6d47885 100644
--- a/meson.build
+++ b/meson.build
@@ -1441,6 +1441,7 @@ vaapi_drm = {
 }
 vaapi_drm += {'use': vaapi['use'] and egl_drm.allowed() and vaapi_drm['deps'].found()}
 if vaapi_drm['use']
+    dependencies += vaapi_drm['deps']
     features += vaapi_drm['name']
 endif
 
@@ -1448,8 +1449,9 @@ vaapi_wayland = {
     'name': 'vaapi-wayland',
     'deps': dependency('libva-wayland', version: '>= 1.1.0', required: get_option('vaapi-wayland')),
 }
-vaapi_wayland += {'use': vaapi['use'] and egl_wayland['use'] and vaapi_wayland['deps'].found()}
+vaapi_wayland += {'use': vaapi['use'] and wayland['use'] and vaapi_wayland['deps'].found()}
 if vaapi_wayland['use']
+    dependencies += vaapi_wayland['deps']
     features += vaapi_wayland['name']
 endif
 
@@ -1464,20 +1466,11 @@ if vaapi_x11['use']
     sources += files('video/out/vo_vaapi.c')
 endif
 
-vaapi_x_egl = {
-    'name': 'vaapi-x-egl',
-    'use': vaapi_x11['use'] and egl_x11.allowed(),
-}
-if vaapi_x_egl['use']
-    features += vaapi_x_egl['name']
-endif
-
 vaapi_egl = {
     'name': 'vaapi-egl',
-    'use': vaapi_x_egl['use'] or vaapi_wayland['use'] or vaapi_drm['use'],
+    'use': egl['use'] or vaapi_drm['use'],
 }
 if vaapi_egl['use']
-    dependencies += [vaapi_wayland['deps'], vaapi_drm['deps']]
     features += vaapi_egl['name']
     sources += files('video/out/hwdec/hwdec_vaapi_gl.c')
 endif
diff --git a/meson_options.txt b/meson_options.txt
index 196119e4d6..1b41693153 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -85,7 +85,6 @@ option('vaapi', type: 'feature', value: 'auto', description: 'VAAPI acceleration
 option('vaapi-drm', type: 'feature', value: 'auto', description: 'VAAPI (DRM/EGL support)')
 option('vaapi-wayland', type: 'feature', value: 'auto', description: 'VAAPI (Wayland support)')
 option('vaapi-x11', type: 'feature', value: 'auto', description: 'VAAPI (X11 support)')
-option('vaapi-x-egl', type: 'feature', value: 'auto', description: 'VAAPI EGL on X11')
 option('vulkan', type: 'feature', value: 'auto', description: 'Vulkan context support')
 option('wayland', type: 'feature', value: 'auto', description: 'Wayland')
 option('x11', type: 'feature', value: 'auto', description: 'X11')
