diff --git a/src/import/import-fs.c b/src/import/import-fs.c.a
index a12ee77..b987361 100644
--- a/src/import/import-fs.c
+++ b/src/import/import-fs.c.a
@@ -1,5 +1,6 @@
 /* SPDX-License-Identifier: LGPL-2.1-or-later */
 
+#include <signal.h>
 #include <getopt.h>
 #include <locale.h>
 
diff --git a/src/import/curl-util.c b/src/import/curl-util.c.a
index d6a16b4..f0ea659 100644
--- a/src/import/curl-util.c
+++ b/src/import/curl-util.c.a
@@ -352,7 +352,6 @@ int curl_header_strdup(const void *contents, size_t sz, const char *field, char
 }
 
 int curl_parse_http_time(const char *t, usec_t *ret) {
-        _cleanup_(freelocalep) locale_t loc = (locale_t) 0;
         const char *e;
         struct tm tm;
         time_t v;
@@ -360,18 +359,14 @@ int curl_parse_http_time(const char *t, usec_t *ret) {
         assert(t);
         assert(ret);
 
-        loc = newlocale(LC_TIME_MASK, "C", (locale_t) 0);
-        if (loc == (locale_t) 0)
-                return -errno;
-
         /* RFC822 */
-        e = strptime_l(t, "%a, %d %b %Y %H:%M:%S %Z", &tm, loc);
+        e = strptime(t, "%a, %d %b %Y %H:%M:%S %Z", &tm);
         if (!e || *e != 0)
                 /* RFC 850 */
-                e = strptime_l(t, "%A, %d-%b-%y %H:%M:%S %Z", &tm, loc);
+                e = strptime(t, "%A, %d-%b-%y %H:%M:%S %Z", &tm);
         if (!e || *e != 0)
                 /* ANSI C */
-                e = strptime_l(t, "%a %b %d %H:%M:%S %Y", &tm, loc);
+                e = strptime(t, "%a %b %d %H:%M:%S %Y", &tm);
         if (!e || *e != 0)
                 return -EINVAL;
 
